lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -58,7 +59,22 @@ export class ClientCredentialClient extends BaseClient {
 
         const cachedAuthenticationResult =
             await this.getCachedAuthenticationResult(request);
+
         if (cachedAuthenticationResult) {
+            // if the token is not expired but must be refreshed; get a new one in the background
+            if (this.lastCacheOutcome === CacheOutcome.REFRESH_CACHED_ACCESS_TOKEN) {
+                this.logger.info(
+                    "ClientCredentialClient:getCachedAuthenticationResult - Cached access token's refreshOn property has been exceeded'. It's not expired, but must be refreshed."
+                );
+
+                // return the cached token, don't wait for the result of this request
+                const accessTokenRefresh = true;
+                this.executeTokenRequest(request, this.authority, accessTokenRefresh);
+            }
+
+            this.lastCacheOutcome = "";
```

Comment:
Please type lastCacheOutcome to the CacheOutcome enum and add a default/empty option if it doesn't already exist